netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -198,7 +213,27 @@ public Runnable start(Listener transportListener) {
      * that it may begin buffering writes.
      */
     b.handler(negotiationHandler);
-    channel = b.register().channel();
+    ChannelFuture regFuture = b.register();
```

Comment:
out of curiosity, why not add a listener?  The future could not be resolved by line 220 right?